Lib/test/test_itertools.py

```diff
@@ -2140,10 +2140,10 @@ class subclass(cls):
                     subclass(*args, newarg=3)
 
         for cls, args, result in testcases:
-            # Constructors of repeat, zip, compress accept keyword arguments.
+            # Constructors of repeat, zip, map, compress accept keyword arguments.
```

Comment:
Is there a clean way to support this?

Reply:
Overriding `__new__`?
```py
                class subclass_with_init(cls):
                    def __init__(self, *args, newarg=None):
                        self.newarg = newarg
                    if cls in [repeat, zip, map, compress]:
                        # Constructors of repeat, zip, map, compress accept keyword arguments.
                        # Their subclasses need an overridden __new__ that ignores
                        # the new keyword arguments.
                        def __new__(cls, *args, newarg=None, **kwargs):
                            return super().__new__(cls, *args, **kwargs)
```

Reply:
I guess that's acceptable. I wasn't sure if it was fine breaking this.

Reply:
IMO, it's fine. If it becomes a problem we should fix all of these.